task/control-plane-agent/src/mgs_gimlet.rs

```diff
@@ -625,20 +625,15 @@ impl SpHandler for MgsHandler {
             component
         }));
 
-        // TODO: Wire up any component info we can (sensor measurements, etc)
-        match component {
-            _ => Err(SpError::RequestUnsupportedForComponent),
-        }
+        self.common.inventory().num_component_details(&component)
     }
 
     fn component_details(
         &mut self,
-        _component: SpComponent,
-        _index: u32,
+        component: SpComponent,
+        index: u32,
     ) -> ComponentDetails {
-        // We never return successfully from `num_component_details()`, so this
-        // function should never be called.
-        panic!()
+        self.common.inventory().component_details(&component, index)
```

Comment:
Same question here: if `index` is untrusted or coming from the network, then i think this could panic.

Reply:
It's trusted here too; the short version is that the devices / components handlers in the big trait are split into "How many do you have?" and "Give me item N". Network requests come in and are validated and/or capped by the value returned by "How many do you have?", so the "Give me item N" handlers are allowed to assume the index is in range.

Reply:
Added comments and a `BoundsChecked` newtype in 73088dc3